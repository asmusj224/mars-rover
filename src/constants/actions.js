export const REQUEST_IMAGE_GALLERY = 'REQUEST_IMAGE_GALLERY';
export const RECEIVE_IMAGE_GALLERY = 'RECEIVE_IMAGE_GALLERY';
export const RECEIVE_IMAGE_GALLERY_ERROR = 'RECEIVE_IMAGE_GALLERY_ERROR';

export const REQUEST_IMAGE_OF_THE_DAY = 'REQUEST_IMAGE_OF_THE_DAY';
export const RECEIVE_IMAGE_OF_THE_DAY = 'RECEIVE_IMAGE_OF_THE_DAY';
export const RECEIVE_IMAGE_OF_THE_DAY_ERROR = 'RECEIVE_IMAGE_OF_THE_DAY_ERROR';