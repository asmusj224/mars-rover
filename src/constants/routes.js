export const IMAGE_OF_THE_DAY = '/image-of-the-day';
export const IMAGE_GALLERY = '/image-gallery';
export const LANDING_PAGE = '/';
