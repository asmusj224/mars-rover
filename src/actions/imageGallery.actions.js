import { REQUEST_IMAGE_GALLERY, RECEIVE_IMAGE_GALLERY, RECEIVE_IMAGE_GALLERY_ERROR } from '../constants/actions';

export const requestImageGallery = () => ({ type: REQUEST_IMAGE_GALLERY });
export const receiveImageGallery = (payload) => ({ type: RECEIVE_IMAGE_GALLERY, payload });
export const receiveImageGalleryError = (payload) => ({ type: RECEIVE_IMAGE_GALLERY_ERROR, payload });