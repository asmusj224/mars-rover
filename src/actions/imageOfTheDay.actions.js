import { REQUEST_IMAGE_OF_THE_DAY, RECEIVE_IMAGE_OF_THE_DAY, RECEIVE_IMAGE_OF_THE_DAY_ERROR } from '../constants/actions';

export const requestImageOfTheDay = () => ({ type: REQUEST_IMAGE_OF_THE_DAY });
export const receieveImageOfTheDay = (payload) => ({ type: RECEIVE_IMAGE_OF_THE_DAY, payload });
export const receieveImageOfTheDayError = (payload) => ({ type: RECEIVE_IMAGE_OF_THE_DAY_ERROR, payload });
