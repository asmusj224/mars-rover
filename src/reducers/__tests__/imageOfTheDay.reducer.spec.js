import reducer from '../imageOfTheDay.reducer';
import * as actions from '../../constants/actions';

const roverImages = [{
    "id": 634105,
    "sol": 1847,
    "camera": {
        "id": 20,
        "name": "FHAZ",
        "rover_id": 5,
        "full_name": "Front Hazard Avoidance Camera"
    },
    "img_src": "http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01847/opgs/edr/fcam/FLB_561457773EDR_F0661516FHAZ00341M_.JPG",
    "earth_date": "2017-10-16",
    "rover": {
        "id": 5,
        "name": "Curiosity",
        "landing_date": "2012-08-06",
        "launch_date": "2011-11-26",
        "status": "active",
        "max_sol": 2134,
        "max_date": "2018-08-07",
        "total_photos": 340100,
        "cameras": [
            {
                "name": "FHAZ",
                "full_name": "Front Hazard Avoidance Camera"
            },
            {
                "name": "NAVCAM",
                "full_name": "Navigation Camera"
            },
            {
                "name": "MAST",
                "full_name": "Mast Camera"
            },
            {
                "name": "CHEMCAM",
                "full_name": "Chemistry and Camera Complex"
            },
            {
                "name": "MAHLI",
                "full_name": "Mars Hand Lens Imager"
            },
            {
                "name": "MARDI",
                "full_name": "Mars Descent Imager"
            },
            {
                "name": "RHAZ",
                "full_name": "Rear Hazard Avoidance Camera"
            }
        ]
    }
}]


describe('Image Of The Day Reducer', () => {
    it('should return intial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });
    it('should handle RECEIVE_IMAGE_OF_THE_DAY', () => {
        expect(reducer([], { type: actions.RECEIVE_IMAGE_OF_THE_DAY, payload: { photos: roverImages } })).toMatchSnapshot();
    });
    it('should handle RECEIVE_IMAGE_OF_THE_DAY_ERROR', () => {
        expect(reducer([], { type: actions.RECEIVE_IMAGE_OF_THE_DAY_Error, payload: [] })).toMatchSnapshot();
    });
});