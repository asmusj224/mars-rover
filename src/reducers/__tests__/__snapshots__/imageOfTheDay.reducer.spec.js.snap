// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Of The Day Reducer should handle RECEIVE_IMAGE_GALLERY_ERROR 1`] = `Array []`;

exports[`Image Of The Day Reducer should handle RECEIVE_IMAGE_OF_THE_DAY 1`] = `
Object {
  "roverCameraName": "Front Hazard Avoidance Camera",
  "roverImage": "http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01847/opgs/edr/fcam/FLB_561457773EDR_F0661516FHAZ00341M_.JPG",
  "roverName": "Curiosity",
}
`;

exports[`Image Of The Day Reducer should handle RECEIVE_IMAGE_OF_THE_DAY_ERROR 1`] = `Array []`;

exports[`Image Of The Day Reducer should return intial state 1`] = `Array []`;
